sdk/tests/deqp/functional/gles3/es3fShaderLoopTests.js

```diff
@@ -0,0 +1,1224 @@
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+'use strict';
+goog.provide('functional.gles3.es3fShaderLoopTests');
+goog.require('framework.common.tcuTestCase');
+goog.require('framework.delibs.debase.deMath');
+goog.require('framework.opengl.gluShaderUtil');
+goog.require('modules.shared.glsShaderRenderCase');
+
+goog.scope(function() {
+	var es3fShaderLoopTests = functional.gles3.es3fShaderLoopTests;
+	var tcuTestCase = framework.common.tcuTestCase;
+	var deMath = framework.delibs.debase.deMath;
+	var gluShaderUtil = framework.opengl.gluShaderUtil;
+	var glsShaderRenderCase = modules.shared.glsShaderRenderCase;
+// Repeated with for, while, do-while. Examples given as 'for' loops.
+// Repeated for const, uniform, dynamic loops.
+
+/**
+ * @enum {number}
+ */
+es3fShaderLoopTests.LoopCase = {
+	LOOPCASE_EMPTY_BODY: 0,								// for (...) { }
+	LOOPCASE_INFINITE_WITH_UNCONDITIONAL_BREAK_FIRST: 1,		// for (...) { break; <body>; }
+	LOOPCASE_INFINITE_WITH_UNCONDITIONAL_BREAK_LAST: 2,		// for (...) { <body>; break; }
+	LOOPCASE_INFINITE_WITH_CONDITIONAL_BREAK: 3,				// for (...) { <body>; if (cond) break; }
+	LOOPCASE_SINGLE_STATEMENT: 4,								// for (...) statement;
+	LOOPCASE_COMPOUND_STATEMENT: 5,							// for (...) { statement; statement; }
+	LOOPCASE_SEQUENCE_STATEMENT: 6,							// for (...) statement, statement;
+	LOOPCASE_NO_ITERATIONS: 7,									// for (i=0; i<0; i++) ...
+	LOOPCASE_SINGLE_ITERATION: 8,								// for (i=0; i<1; i++) ...
+	LOOPCASE_SELECT_ITERATION_COUNT: 9,						// for (i=0; i<a?b:c; i++) ...
+	LOOPCASE_CONDITIONAL_CONTINUE: 10,							// for (...) { if (cond) continue; }
+	LOOPCASE_UNCONDITIONAL_CONTINUE: 11,						// for (...) { <body>; continue; }
+	LOOPCASE_ONLY_CONTINUE: 12,									// for (...) { continue; }
+	LOOPCASE_DOUBLE_CONTINUE: 13,								// for (...) { if (cond) continue; <body>; continue; }
+	LOOPCASE_CONDITIONAL_BREAK: 14,								// for (...) { if (cond) break; }
+	LOOPCASE_UNCONDITIONAL_BREAK: 15,							// for (...) { <body>; break; }
+	LOOPCASE_PRE_INCREMENT: 16,									// for (...; ++i) { <body>; }
+	LOOPCASE_POST_INCREMENT: 17,								// for (...; i++) { <body>; }
+	LOOPCASE_MIXED_BREAK_CONTINUE: 18,
+	LOOPCASE_VECTOR_COUNTER: 19,								// for (ivec3 ndx = ...; ndx.x < ndx.y; ndx.x += ndx.z) { ... }
+	LOOPCASE_101_ITERATIONS: 20,								// loop for 101 iterations
+	LOOPCASE_SEQUENCE: 21,										// two loops in sequence
+	LOOPCASE_NESTED: 22,										// two nested loops
+	LOOPCASE_NESTED_SEQUENCE: 23,								// two loops in sequence nested inside a third
+	LOOPCASE_NESTED_TRICKY_DATAFLOW_1: 24,						// nested loops with tricky data flow
+	LOOPCASE_NESTED_TRICKY_DATAFLOW_2: 25						// nested loops with tricky data flow
```

Comment:
Are all of these loop variants expected to pass? How does the native test harness do on these tests? Have you noticed any loop constructs in the tests whose behavior is outside the scope of the GLSL ES 3.00 spec?